jsonrpc/server_test.go

```diff
@@ -0,0 +1,147 @@
+	}{
+		{
+			req: `{]`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32600,"message":"invalid character ']' looking for beginning of object key string"},"id":null}`,
+		},
+		{
+			req: `{"jsonrpc" : "1.0", "id" : 1}`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32600,"message":"unsupported RPC request version"},"id":null}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "doesnotexits" , "id" : 2}`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32601,"message":"method not found"},"id":2}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method", "params" : [3, false] , "id" : 3}`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32602,"message":"missing param in list"},"id":3}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method", "params" : [3, false, "error message"] , "id" : 3}`,
+			res: `{"jsonrpc":"2.0","result":{"doubled":6},"id":3}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method", "params" : [3, true, "error message"] , "id" : 4}`,
+			res: `{"jsonrpc":"2.0","error":{"code":44,"message":"error message"},"id":4}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5, "shouldError" : false, "msg": "error message" } , "id" : 5}`,
+			res: `{"jsonrpc":"2.0","result":{"doubled":10},"id":5}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5 } , "id" : 5}`,
+			res: `{"jsonrpc":"2.0","result":{"doubled":10},"id":5}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5, "shouldError" : true } , "id" : 22}`,
+			res: `{"jsonrpc":"2.0","error":{"code":44,"message":""},"id":22}`,
+		},
+		{
+			req: `{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "shouldError" : true } , "id" : 22}`,
+			res: `{"jsonrpc":"2.0","error":{"code":-32602,"message":"missing non-optional param"},"id":22}`,
+		},
+		{
+			req: `[]`,
+			res: `[]`,
+		},
+		{
+			req: `[{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5 } , "id" : 5}]`,
+			res: `[{"jsonrpc":"2.0","result":{"doubled":10},"id":5}]`,
+		},
+		{
+			req: `[{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5 } , "id" : 5},
+					{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 44 } , "id" : 6}]`,
+			res: `[{"jsonrpc":"2.0","result":{"doubled":10},"id":5},{"jsonrpc":"2.0","result":{"doubled":88},"id":6}]`,
+		},
+		{
+			req: `[{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 5 } , "id" : 5},
+					{"jsonrpc" : "2.0", "method" : "fail",
+					"params" : { "num" : 5 } , "id" : 7},
+					{"jsonrpc" : "2.0", "method" : "method",
+					"params" : { "num" : 44 } , "id" : 6}]`,
+			res: `[{"jsonrpc":"2.0","result":{"doubled":10},"id":5},{"jsonrpc":"2.0","error":{"code":-32601,"message":"method not found"},"id":7},{"jsonrpc":"2.0","result":{"doubled":88},"id":6}]`,
+		},
+	}
+
+	for _, test := range tests {
+		res, err := server.Handle([]byte(test.req))
+		assert.NoError(t, err)
```

Comment:
```suggestion
		require.NoError(t, err)
```

Otherwise, there will be a panic on the next line is `res` is nil.